language/tools/move-cli/src/main.rs

```diff
@@ -198,24 +210,109 @@ fn interface_files_dir(build_dir: &str) -> Result<String> {
     Ok(dir)
 }
 
+fn shadow_storage(
```

Comment:
The compiler has no notion of generated files (as an argument), so it would be a bit weird to have something to filter over that. 

I think this current function only makes sense in the CLI where it is managing a build directory

Reply:
I guess I mean: pass some source files S and some bytecode files B to the compiler. Some modules may be defined in both S and B, but we want to ignore the copy in B and compile the one in S.

Reply:
Ah... yeah I am unsure how common that will be. Definitely worth considering moving it if it seems like a more common usecase

Reply:
Now that I'm thinking about it, I wonder if this should just be the default behavior between source and lib files...